lemarche/siaes/models.py

```diff
@@ -496,37 +497,63 @@ def filter_with_tender_tendersiae_status(self, tender, tendersiae_status=None):
 
         return qs.distinct()
 
-    def with_tender_stats(self):
+    def with_tender_stats(self, days=0):
         """
-        Enrich each Siae with stats on their linked Tender
+        Enrich each Siae with stats on their linked Tender.
+        Optionally, limit the stats to the last `days` days.
         """
+        date_limit = timezone.now() - timedelta(days=days) if days > 0 else None
+        date_filter = Q() if date_limit is None else Q(tendersiae__email_send_date__gte=date_limit)
+
         return self.annotate(
-            tender_count_annotated=Count("tenders", distinct=True),
+            tender_count_annotated=Count("tenders", distinct=True),  # noqa
             tender_email_send_count_annotated=Sum(
-                Case(When(tendersiae__email_send_date__isnull=False, then=1), default=0, output_field=IntegerField())
+                Case(
```

Comment:
Alors ici, je pense que tu peux le faire directement sans variable "date_filter" comme tu as fait plus bas.

Aussi, je pense que tu peux faire soit l'un soit l'autre et que tu n'as pas besoin du &, car c'est sur le même champs date et si tu es gte à une date, c'est par null.

```suggestion
                Case(
                    When(
                        (
                            Q(tendersiae__email_send_date__isnull=False)
                            if date_limit is None
                            else Q(tendersiae__email_send_date__gte=date_limit)
                        ),
                        then=1,
                    ),
```